
Due to the way that neovim bundles luajit, it ends up with a lua
package.path and package.cpath including the .deps directory under
the neovim source, which is unhelpful in the deployed package.

To fix this we split the build into a build and install step and
use the proper OOCE prefix during build. We also switch to using the
`amalg` target here as recommended by the luajit project for any
binary distributions.

diff -wpruN --no-dereference '--exclude=*.orig' a~/cmake.deps/cmake/BuildLuajit.cmake a/cmake.deps/cmake/BuildLuajit.cmake
--- a~/cmake.deps/cmake/BuildLuajit.cmake	1970-01-01 00:00:00
+++ a/cmake.deps/cmake/BuildLuajit.cmake	1970-01-01 00:00:00
@@ -44,7 +44,12 @@ if(APPLE)
 endif()
 
 if(UNIX)
-  BuildLuaJit(INSTALL_COMMAND ${BUILDCMD_UNIX}
+  BuildLuaJit(
+    BUILD_COMMAND ${BUILDCMD_UNIX}
+    CC=${DEPS_C_COMPILER} PREFIX=/opt/ooce
+    ${DEPLOYMENT_TARGET} amalg
+
+    INSTALL_COMMAND ${BUILDCMD_UNIX}
     CC=${DEPS_C_COMPILER} PREFIX=${DEPS_INSTALL_DIR}
     ${DEPLOYMENT_TARGET} install)
 
